docs/Upgrade-Guide/upgrade_to_6.md

```diff
@@ -0,0 +1,81 @@
+## Upgrade procedure to Gluster 6, from Gluster 5.x, 4.1.x, and 3.12.x
+
+We recommend you to read [Release notes of v6.0](https://github.com/gluster/glusterfs/blob/v6.0/doc/release-notes/6.0.md) to be aware of the features, and fixes in the release before the upgrade.
+
+> **NOTE:** Upgrade procedure remains the same as with 4.1.x release
+
+Refer, to the [Upgrading to 4.1](./upgrade_to_4.1.md) guide and follow
+documented instructions, replacing 6 when you encounter 4.1 in the guide as the
+version reference.
+
+## Deprecated translator upgrade guide
+
+With glusterfs-6.0 release, we are planning to deprecate few of the
+features (or volume types) which existed before.
+
+If you have any one of the below volume types, please go through
+instructions below, so you can preserve sanity in your glusterfs
+release.
+
+### Stripe volume
+
+Stripe was providing a way to divide the data, and store it across
+different bricks. This translator used feature of filesystem 'holes',
+ie, a seek() and write() method to handle namespace and 'stat()'
+challenges.
+
+We now have a feature called `shard` which takes care of the similar
+usecases, and is present in code-base since last 4 years.
+
+Now, if you are using this translator/volume type, there is no
+online upgrade possible. You have to create one more glusterfs
+volume (of different volume type which is supported) and migrate
+data on top of the mount point (like any other rsync like tool),
+to the new volume mount point.
+
+Once you copy over the data, you can delete the volume, and now you
+are set for the upgrade to `glusterfs-6.0`.
+
+
+### Tier volume
+
+If you are using Tier feature, this is not supported in glusterfs-6.0.
+
+So, please make sure you detach the tier bricks, and convert the volume
+to replica/distribute or disperse (without tier), and once all the
+migration is complete, you will be ready for upgrade.
+
+
+### Misc
+
+* BD xlator
+* glupy
+* experimental/*
+
+All the above translators were not supported in previous versions, but
+users had an option to create the volumes. Now, if you had created these
+volumes, please delete these volumes and then upgrade to glusterfs-6.0
```

Comment:
migrate data out of these volumes, not delete the volumes :)

Reply:
Done